simpleflow/workflow.py

```diff
@@ -27,7 +31,21 @@ def submit(self, func, *args, **kwargs):
             :rtype: Future.
 
         """
-        return self._executor.submit(func, *args, **kwargs)
+        # If the func is a child workflow, call directly
+        # the executor
+        if inspect.isclass(func) and issubclass(func, Workflow):
+            return self._executor.submit(func, *args, **kwargs)
+        elif isinstance(func, activity.Activity):
+            return self._executor.submit(func.activity_type, *func.args, **func.kwargs)
+        elif isinstance(func, activity.ActivityType):
+            return self._executor.submit(func, *args, **kwargs)
+        elif isinstance(func, (canvas.Group, canvas.Chain)):
+            return func.submit(self._executor)
```

Comment:
Why not leaving that responsibility to the executor? Feels weird to have two different submission methods, and I'm not sure I see the gain.